File: Modules/ServerTypes/Master/MasterPostcodeForm.ts
import { LookupEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface MasterPostcodeForm {
    MasterCity: LookupEditor;
    MasterState: LookupEditor;
    PostCode: StringEditor;
}

export class MasterPostcodeForm extends PrefixedContext {
    static formKey = 'Master.MasterPostcode';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MasterPostcodeForm.init)  {
            MasterPostcodeForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;

            initFormType(MasterPostcodeForm, [
                'MasterCity', w0,
                'MasterState', w0,
                'PostCode', w1
            ]);
        }
    }
}

File: Modules/HumanResource/Master/MasterPostcode/MasterPostcodeForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Master.Forms;

[FormScript("Master.MasterPostcode")]
[BasedOnRow(typeof(MasterPostcodeRow), CheckNames = true)]
public class MasterPostcodeForm
{
    public int MasterCity { get; set; }
    public int MasterState { get; set; }
    public string PostCode { get; set; }
}